plugins/modules/ec2_security_group.py

```diff
@@ -1170,6 +1192,7 @@ def get_final_rules(client, module, security_group_rules, specified_rules, purge
                             rule[source_type] = [rule[source_type]]
                         format_rule[rule_key] = [{source_type: target} for target in rule[source_type]]
             if rule.get('group_id') or rule.get('group_name'):
+                # XXX bug - doesn't cope with a list of ids/names
```

Comment:
Existing bug - https://github.com/ansible-collections/amazon.aws/issues/1217 want to split proper solution as the diff code is gnarly